packages/database/src/core/Repo.ts

```diff
@@ -464,17 +465,37 @@ export function repoGetValue(repo: Repo, query: QueryContext): Promise<Node> {
   if (cached != null) {
     return Promise.resolve(cached);
   }
+  const tag = syncTreeRegisterQuery(repo.serverSyncTree_, query);
   return repo.server_.get(query).then(
     payload => {
-      const node = nodeFromJSON(payload as string).withIndex(
+      const node = nodeFromJSON(payload).withIndex(
         query._queryParams.getIndex()
       );
-      const events = syncTreeApplyServerOverwrite(
-        repo.serverSyncTree_,
-        query._path,
-        node
-      );
-      eventQueueRaiseEventsAtPath(repo.eventQueue_, query._path, events);
+      // if this is a filtered query, then overwrite at path
+      if (query._queryParams.loadsAllData()) {
+        syncTreeApplyServerOverwrite(repo.serverSyncTree_, query._path, node);
+      } else {
+        // Simulate `syncTreeAddEventRegistration` without events/listener setup.
+        // We do this (along with the syncTreeRemoveEventRegistration` below) so that
+        // `repoGetValue` results have the same cache effects as initial listeners
+        // updates.
+        syncTreeApplyTaggedQueryOverwrite(
+          repo.serverSyncTree_,
+          query._path,
+          node,
+          tag
+        );
+        // Call `syncTreeRemoveEventRegistration` with a null event registration, since there is none.
+        const cancels = syncTreeRemoveEventRegistration(
+          repo.serverSyncTree_,
+          query,
+          null
+        );
+        assert(
```

Comment:
We should never get cancel events here, but maybe we should turn this into a warning log just in case.

```
if (cancels.length > 0) {
  repoLog(repo, 'ignoring unexpected cancel events in get for query ' + stringify(query));
}
```